modes/mu4e-views/evil-collection-mu4e-views.el

```diff
@@ -0,0 +1,189 @@
+;; (defun evil-collection-mu4e-views-set-state ()
+;;   "Set the appropriate initial state of all mu4e modes."
+;;   (dolist (mode '(mu4e-main-mode
+;;                   mu4e-headers-mode
+;;                   mu4e-view-mode
+;;                   mu4e-org-mode))
+;;     (evil-set-initial-state mode 'normal))
+;;   (evil-set-initial-state 'mu4e-compose-mode 'insert))
+
+;;; Define bindings
+
+;; TODO: Inhibit insert-state functions as per Evil Collection.
+(defvar evil-collection-mu4e-views-mode-map-bindings
+  `((mu4e-headers-mode-map
+     "v" mu4e-views-mu4e-select-view-msg-method
+     ,(kbd "M-b") mu4e-views-cursor-msg-view-window-up
+     ,(kbd "M-f") mu4e-views-cursor-msg-view-window-down)
+
+    (mu4e-views-view-actions-mode-map
+     " " mu4e-view-scroll-up-or-next
+     [tab] shr-next-link
+     [backtab] shr-previous-link
+     "q" mu4e-views-mu4e-headers-windows-only
+     "gx" mu4e-views-mu4e-view-go-to-url
+     "gX" mu4e-views-mu4e-view-fetch-url
+     "C" mu4e-compose-new
+     ;; "H" mu4e-view-toggle-html
+     ;; "E"               mu4e-compose-edit
+     ;; "F"               mu4e-compose-forward
+     "R" mu4e-compose-reply
+     "cc" mu4e-compose-new
+     "ce" mu4e-compose-edit
+     "cf" mu4e-compose-forward
+     "cr" mu4e-compose-reply
+     "p" mu4e-views-mu4e-view-save-attachment
+     ;; "P" mu4e-view-save-attachment-multi ; Since mu4e 1.0, -multi is same as normal.
+     "O" mu4e-headers-change-sorting
+     "o" mu4e-views-mu4e-view-open-attachment
+     ;; "A" mu4e-view-attachment-action
+     "a" mu4e-views-mu4e-view-action
+     "J" mu4e~headers-jump-to-maildir
+     ;; "[[" mu4e-view-headers-prev-unread
+     ;; "]]" mu4e-view-headers-next-unread
+     ;; "gk" mu4e-view-headers-prev-unread
+     ;; "gj" mu4e-view-headers-next-unread
+     "\C-j" mu4e-views-mu4e-headers-next
+     "\C-k" mu4e-views-mu4e-headers-prev
+     "x" mu4e-view-marked-execute
+     "&" mu4e-view-mark-custom
+     "*" mu4e-view-mark-for-something   ; TODO: Don't override "*".
+     "m" mu4e-view-mark-for-move
+     "r" mu4e-view-mark-for-refile
+     "D" mu4e-view-mark-for-delete
+     "d" mu4e-view-mark-for-trash
+     "=" mu4e-view-mark-for-untrash
+     "u" mu4e-view-unmark
+     "U" mu4e-view-unmark-all
+     "?" mu4e-view-mark-for-unread
+     "!" mu4e-view-mark-for-read
+     "%" mu4e-view-mark-pattern
+     "+" mu4e-view-mark-for-flag
+     "-" mu4e-view-mark-for-unflag
+     "zr" mu4e-headers-toggle-include-related
+     "zt" mu4e-headers-toggle-threading
+     ;; "za" mu4e-view-toggle-hide-cited
+     "gl" mu4e-show-log
+     "s" mu4e-view-search-edit
+     ;; "|" mu4e-view-pipe
+     ;; "." mu4e-view-raw-message
+     ,(kbd "C--") mu4e-headers-split-view-shrink
+     ,(kbd "C-+") mu4e-headers-split-view-grow
+     "T" (lambda ()
+           (interactive)
+           (mu4e-headers-mark-thread nil '(read)))
+     ,@(when evil-want-C-u-scroll
+         '("\C-u" evil-scroll-up))))
+  "All evil-mu4e-views bindings.")
```

Comment:
This looks copy-pasted from evil-collection-mu4e.  Instead, I suggest you inherit from the mu4e bindings, this way this whole file can be much shorter.